docs/development/creating-editing-renovate-presets.md

```diff
@@ -0,0 +1,46 @@
+# Creating/editing Renovate presets
+
+Renovate comes with default presets that you can find in the `lib/config/presets/internal` directory.
+You can suggest changes to the presets with a pull request.
+
+Follow the rules below to increase the chance that your pull request gets merged.
+
+## General rules
+
+1. Only create/edit presets for problems which can not be fixed upstream
+1. The internal preset should be helpful for a significant number of Renovate users
+
+### Specific rules
+
+#### Group presets
+
+We have multiple kinds of `group:` presets, with different rules.
+
+##### Rules for `group:monorepo` preset
+
+1. Only group dependencies that _must_ be updated together
+
+##### Rules for `group:recommended` presets
+
+1. The `group:recommended` preset is for related dependencies which aren't from a monorepo but which usually need to be updated together as separate PRs may each break
+
+##### Rules for `group:*` presets
+
+1. Finally, any other `group:*` presets can be added if they are beneficial to a wide number of users
+1. They don't need to be added to `group:recommended`, meaning that users will "opt in" to them one by one and not get them automatically from `config:base`, which includes `group:monorepo` and `group:recommended`
+
+#### Replacement presets
+
+Rules:
+
+1. Replacement PRs should ideally propose a replacement only once the user is on a compatible version, by specifying a compatible `matchCurrentVersion` constraint
+1. If no compatible replacement upgrade is possible, it's acceptable to propose an incompatible one (e.g. a major version upgrade)
+1. Replacements should update the user to the first recommended version of the new dependency and not include any new changes - whether breaking or not - if they can be avoided
```

Comment:
Wouldn't rule 1 and 3 cause a three-stage approach?:

1. Update to the latest version of the initial package
2. Switch to the first version of the replacement package
3. Update to the latest version of the replacement package

So for example: `ant:ant:1.4.1` => `ant:ant:1.7.0` => `org.apache:ant:1.7.0` => `org.apache.ant:ant:1.10.12`

Would that be the preferable outcome of replacements?

Reply:
Yes, in general it would cause that. But we can be flexible if someone thinks there's a better way. What I really want to avoid is *breaking* replacement changes when a smoother alternative is possible

Reply:
That is what we want!

We want users to:

- First be within the latest `major` range of the old package
- Then they'll get an `replacement` PR from Renovate to update them to the earliest possible version of the replacement package
- After that is merged, they'll get Renovate PRs for updates as normal.

Reply:
Are you happy with the current text then? I'm not sure if I need to edit the text or not based on this discussion.